specs/service_deduction.adoc

```diff
@@ -0,0 +1,174 @@
+// vim: tw=79
+
+= Service Deduction Module for Node Agent
+
+Node agents need to ship with modules that can detect various services already
+present on the nodes
+
+== Problem description
+
+This specification describes service deduction module to manage the services
+present on the nodes and populate the details; to and from the central store.
+The list could then be presented to the administrator via the API and the UI
+to allow selection of the nodes for monitoring, service operation and
+troubleshoting.
+
+== Use Cases
+
+* Tendrl should detects the service details of the storage node and show
+the details and pushes the details to central store for further use.
+
+* Use Case1: Monitoring
+  ** The list of available nodes presented to the administrator via APIs
+  ** User selects one node in the cluster
+  ** From the selected node user selects the deducted services
+  ** From the selected service user can watch the status
+     or submit request to start or stop or reload the service
+
+* Use Case2: Trouble-shooting
+  ** The list of available nodes presented to the administrator via APIs
+  ** User selects one node in the cluster
+  ** From the selected node user selects the deducted services
+  ** From the selected service user submit request to start or stop or
+     reload the service
+
+* Use Case3: During import
+  ** The list of available nodes presented to the administrator via APIs
+  ** User selects one node in the cluster
+  ** Use the selected node to extract the details of the cluster which
+     uses deducted service status
+
+== Proposed change
+
+Node Agents related changes:
+* Add a module which fetches the service details like service-name, execution
```

Comment:
Services or Service details?

Can detect various services already present on the nodes. This includes, but is not limited to:
    ceph services
    gluster services
    tendrl api
    tendrl monitoring application
    tendrl central store